examples/simple-state/koka/state.kk

```diff
@@ -0,0 +1,74 @@
+// A state effect that is polymorphic in the state of type `a`
+effect state<a> {
+  fun get() : a
+  fun put( x : a) : ()
+}
+
+// We define first a state handler using functions.
+// But you would never do this in practice in Koka! --
+// use a parameterised handler instead: see `fun state` for that.
+//
+// inferred type: forall<a,b> (init : a, action : () -> state<a> b) -> b
+fun fstate(init,action) {
+  (handle(action) {
+    get()    -> fun(st){ resume(st)(st) }
+    put(x)   -> fun(st){ resume(())(x) }
+    return x -> fun(_st){ x }
+  })(init)
+}
+
+
+fun example1a() {
+  // In koka an anonymous function without arguments can be written
+  // with just braces (and passed outside the argument parenthesis)
+  fstate(10) {
+    val x = get()
+    put(4 + x)
+    3*get()
+  }
+}
+
+fun example1b() {
+  // Koka has `using` syntactic sugar to make the rest of the scope
+  // the anonymous function argument, and this is equivalent to the
+  // previous example.
+  using fstate(10)
+  val x = get()
+  put(4 + x)
+  3*get()
+}
+
+fun example2() 
+{
+  fun add-to(x : int) { 
+    put(get() + x) 
+  }
+  using fstate(0)
+  for(1,100,add-to)
+  get()
+}
+
+// Here we use a paremeterised handler with a local state `st`
```

Comment:
I disagree. We should be comparing idiomatic code from each language. The idiomatic version of simple state in Koka and Links uses parameterised handlers. The idiomatic version of simple state in Frank uses shallow handlers. The description of the example should be flexible enough (and perhaps updated) in order to accommodate different idioms. I think it is more valuable to highlight the differences in features between languages rather than stressing the lowest common denominator.

Reply:
There should be multiple examples: idiomatic usage, as well as trying to impose the same structure in multiple languages, with a careful readme that helps the reader navigate through them.

You well know that one learns much by trying to faithfully express one collection of abstractions using another ;).

Reply:
That certainly makes sense sometimes, but the idiomatic version should be the primary one. Daan did include the crippled version in Koka too. For Frank it is simply not possible to express the Eff version because Frank does not have deep handlers (and I'm sure the same issue will crop up with other languages and features).

Reply:
Great! Then there should be a description of Frank's inability to do so with deep handlers. This kind of comparison is exactly the kind of examples and non examples this repo is for.

Similarly, there should be a corresponding shallow handler example in effect demonstrating how awkward it is in Eff.

Reply:
That's too fine a level of granularity for my taste.

Reply:
You misunderstand. There is absolutely a place for the kind of commentary you describe. But, placing each minor variant of state in a different directory (rather than in the same file, as @daanx did), is too fine a level of granularity for my taste. More generally, if we are too fussy then people won't bother submitting any examples. I think it would be better to err on the side of accepting mildly ill-formed examples rather than rejecting them and then losing the goodwill of the submitters. We can always curate them and add further commentary after the fact.

Reply:
Maybe following the example of [Rosettacode](http://rosettacode.org) might be useful here? They seem to allow for [having a few different implementations per language](http://rosettacode.org/wiki/Sorting_algorithms/Quicksort#Haskell), if the different versions help demonstrate the interesting qualities of the language.

Reply:
I agree with @slindley here: for each example problem, first provide the idiomatic version, and then, if applicable, also provide another approach. When I am looking to compare, I would like to see the idiomatic versions first, not crippled ones just for the sake of emulating other approaches. 
(This also helps language designers: otherwise we need at least 2 versions for each example, shallow for Frank and deep for Koka.)

I think it is better to provide higher level challenge _problems_ instead of requiring the exact same language _features_. For example, do an example with producer/consumer where Frank can shine with shallow handlers + multi handlers, where Koka would need to get by in some other way; that is really interesting to compare from my perspective.

Anyway, just my 2c, I am already very grateful that this repository is set up.